File: 0739-daily-temperatures/0739-daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] t) {
        int size = t.length;
        int[] res = new int[size];
        Stack<Integer> stk = new Stack<>();
        for(int i=0; i<size; i++){
            while(!stk.isEmpty() && t[i] > t[stk.peek()]){
                int cnt = stk.pop();
                res[cnt] = i-cnt;
            }
            stk.push(i);
        }
        
        return res;
        
    }
}

File: 0739-daily-temperatures/0739-daily-temperatures.py
class Solution:
    def dailyTemperatures(self, t: List[int]) -> List[int]:
        res = [0]*len(t)
        stk = []
        for i in range(len(t)):
            while(stk and t[i]>t[stk[-1]]):
                cnt = stk.pop()
                res[cnt] = i - cnt
            stk.append(i)
        return res